### PR TITLE
Polish bundle: #197 appliance delete cascade + #165 GUI timezone + #280 MCP read tools

### LINKED ISSUE #165: appliance: timezone change post-install via GUI
## Problem

Installer wizard asks for timezone (IANA name) at install time. No post-install GUI to change it short of `timedatectl set-timezone` over SSH. Trivial but real — sites move, operators travel, hardware gets shipped to a different DC.

## Schema sketch

`platform_settings.timezone` (string, IANA name; default UTC).

## Acceptance

- [ ] Backend GET/PUT
- [ ] Runner does `timedatectl set-timezone <iana>` on the host
- [ ] Frontend dropdown with IANA names + filter
- [ ] Audit log carries timezone change

### LINKED ISSUE #197: [feat] Appliance delete should cascade to its DNS/DHCP server rows
### Area

Appliance / Fleet

### What problem does this solve?

When an operator deletes an Application appliance from `/appliance → Fleet`, the appliance row itself goes away (`Appliance` + cascading `PairingClaim` + cert) but the `dns_server` / `dhcp_server` rows the supervisor registered as part of role assignment stay attached to their server groups. The orphaned rows show as offline (no heartbeat) but never get pruned — operators have to remember to manually delete them from **DNS → Server Groups** / **DHCP → Server Groups** after every delete.

Worse: the orphaned rows still get ConfigBundle long-polled by an absent agent (eats a Postgres connection slot per stale row), still get zone-state heartbeats sent to them (404s log a warning every minute), and still get counted against group quorum for HA Kea pairs.

### Proposed solution

On `DELETE /api/v1/appliance/{id}`, before dropping the `Appliance` row, identify and delete any `dns_server` / `dhcp_server` rows that belong to this appliance.

Linkage candidates (current schema):

- **Hostname match.** `Appliance.hostname` ↔ `DNSServer.hostname` / `DHCPServer.hostname`. Most direct, but operators sometimes rename. Add a one-shot `Appliance.created_dns_server_ids` / `Appliance.created_dhcp_server_ids` JSONB tracking column.
- **Cert fingerprint.** The supervisor signs every register call with its mTLS cert. We could record the cert fingerprint on the `dns_server` / `dhcp_server` row at register time, then match on delete. More robust than hostname but requires a schema migration.
- **`last_seen_ip` match.** Works in most cases but breaks when the supervisor's IP changes between register and delete.

Recommend the **dedicated FK** approach: add a nullable `Appliance.id` FK on `dns_server` + `dhcp_server`, populated at supervisor-driven register time. `ON DELETE CASCADE` does the rest — Postgres cleans up automatically when the appliance row goes away.

For existing rows registered pre-fix (no FK populated), the delete path can fall back to hostname match with an explicit operator-visible "Will also delete the following server rows: ..." confirmation step before commit.

### Acceptance criteria

- [ ] New `appliance_id` nullable FK on `dns_server` + `dhcp_server` with `ON DELETE CASCADE` (alembic migration).
- [ ] Supervisor's `apply_role_assignment` path passes the supervisor's `appliance_id` through to the service container's register call so the new column gets populated automatically.
- [ ] `DELETE /api/v1/appliance/{id}` cascades via the new FK + as a fallback also sweeps hostname-matched rows for legacy installs.
- [ ] Fleet UI delete confirmation modal lists the dependent `dns_server` / `dhcp_server` rows that will be removed so the operator knows the full blast radius before clicking.
- [ ] Audit-log row on delete records the count + names of cascaded rows.
- [ ] Test coverage: delete appliance → verify zero orphan `dns_server` / `dhcp_server` rows remain.

### Alternatives considered

- **Soft-delete the dns/dhcp server rows instead** so an accidentally-deleted appliance can be restored. Rejected — there's no restore flow for `Appliance` itself (delete is permanent today), so coupling the server rows the same way is consistent. Operators who need restore should `git revert` from a backup.
- **Leave the rows + just mark them stale.** Rejected — the rows are tied to a now-gone Ed25519 identity, so they can't ever re-register cleanly. Cleanup is the right shape.

### Cross-references

Related to #170 Wave C2 (multi-role assignment). The supervisor → service container key-passing pipeline that's already wired up post-#170 is the natural place to thread the new `appliance_id` into the register call.

### PR DESCRIPTION
**Pull request by mzac**
Closes #197, closes #165. Partial close of #280 (read tools shipped here; the 17 `propose_*` writes deferred to **#304** because each needs an `Operation` class — substantial enough to be its own PR).

### #197 — Appliance delete cascades to DNS/DHCP server rows ✅

Operators deleting an appliance from Fleet no longer leave ghost DNS / DHCP server rows behind. Each Application appliance's `dns_server` / `dhcp_server` rows are now linked back via a nullable `appliance_id` FK with `ON DELETE CASCADE`, and the soft-delete endpoint sweeps the dependent rows in the same transaction.

- Migration `b1d4c9e57f02` — `appliance_id` FK on both server tables.
- Models: matching ORM columns.
- New `GET /appliances/{id}/dependents` endpoint — previews the rows the delete flow will sweep.
- `DELETE /appliances/{id}` now removes the dependents + audits the cascaded names.
- Frontend delete-confirm modal lists the dependent rows (matched by `appliance_id` OR `hostname`) so the operator sees the full blast radius before clicking Revoke.
- Tests: dependents preview, soft-delete cascade, zero-dependent edge case.
- **Deferred to follow-up**: wiring `appliance_id` at supervisor-driven register time. Today every dependent row is found via hostname-match, which works for legacy AND newly-registered rows. The FK column is forward-compatible.

### #165 — Timezone change post-install via GUI ✅

Full end-to-end pipeline from operator click → host `timedatectl`.

- Migration `c2e6a89d4b15` — `platform_settings.timezone` (empty = "follow install-time default").
- Backend: `SettingsResponse.timezone` + `SettingsUpdate.timezone` with `zoneinfo.ZoneInfo` field-validator (bad IANA names 422 before reaching the host runner).
- `SupervisorHeartbeatResponse.desired_timezone` carries the operator-set value.
- Supervisor `appliance_state.maybe_fire_timezone` writes `/var/lib/spatiumddi-host/release-state/tz-pending` when current and desired diverge.
- Host-side `spatiumddi-tz-reload` bash runner + `.path` + `.service` units. Validates against `timedatectl list-timezones` + applies via `timedatectl set-timezone`. Status sidecar surfaces failures.
- `mkosi.postinst` enables the new `.path` unit alongside the chrony / SNMP / firewall units.
- Frontend `TimezoneSection` inside the NTP tab. Text input backed by a `<datalist>` of `Intl.supportedValuesOf("timeZone")` (modern browsers) with a curated fallback for old browsers.

### #280 (partial) — Operator Copilot read tools ✅

7 new MCP read tools across 3 modules. Each surface gets default-enabled reads; the write tools (`propose_*`) are deferred to **#304** because each needs an `Operation` class with `preview()` + `apply()` methods (~50-80 lines × 17 tools).

- **Conformity** (NEW `tools/conformity.py`): `list_conformity_policies`, `find_conformity_results`, `get_conformity_summary`.
- **Webhooks** (NEW `tools/webhooks.py`): `list_webhooks` (secrets NEVER returned — `secret_set` boolean only), `get_webhook_event_types`, `find_webhook_deliveries`.
- **DNSSEC** (extended `tools/dns.py`): `find_zone_dnssec_info` (DS records + sync timestamp).

## Out of scope / deferred

- **#197 follow-up**: wire `appliance_id` into supervisor's register flow so new rows get the FK populated. Today hostname-match handles every case; the FK column is forward-compatible.
- **#280 → #304**: all 17 `propose_*` write tools (conformity / webhooks / backup / DNSSEC / DNS-import / multicast-domain / SNMP / NTP). Each needs an Operation class — substantial enough to warrant its own PR.

## Test plan

- `make ci`: ✅
- Venv-pinned `.venv/bin/{ruff,black,mypy}` on full repo (470 source files): ✅
- `tests/test_appliance_delete_dependents.py` + `tests/test_health.py` in dev container: 5/5 ✅
- **Real-hardware validation needed** post-merge: install fresh appliance ISO, exercise the delete-confirm modal + the timezone picker flow + verify the new MCP tools surface in the Operator Copilot's Tool Catalog.

## Linked issues

- Closes #197
- Closes #165
- Partial close of #280 (reads ✅; writes → #304)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: backend/app/api/v1/appliance/supervisor.py [hunk 1]
```diff
@@ -1972,6 +1981,121 @@ class DeleteApplianceRequest(BaseModel):
...
+# wiring didn't catch).
+
+
+class AppliancedDependentServer(BaseModel):
+    """One DNS or DHCP server row tied to an appliance about to be
+    deleted. Surfaced to the operator in the delete-confirm modal so
+    they see the full blast radius before clicking."""
+
+    kind: str  # "dns" | "dhcp"
+    id: uuid.UUID
```
[the remark this answers is not in the document]
Comment (mzac) [reply backend/app/api/v1/appliance/supervisor.py]: Fixed in 2c96105 — renamed to `ApplianceDependentServer` (typo). Response model + the local list-comprehensions in `_find_appliance_dependents` both updated.

File: backend/app/api/v1/appliance/supervisor.py [hunk 2]
```diff
@@ -1972,6 +1981,121 @@ class DeleteApplianceRequest(BaseModel):
...
+    # appliance's hostname (see ``register`` in dns/agents.py).
+    dns_rows = (
+        (
+            await db.execute(
+                select(DNSServer).where(
+                    or_(
+                        DNSServer.appliance_id == appliance.id,
+                        DNSServer.host == appliance.hostname,
+                        DNSServer.name == appliance.hostname,
+                    )
```
[the remark this answers is not in the document]
Comment (mzac) [reply backend/app/api/v1/appliance/supervisor.py:2041]: Fixed in 2c96105 — dropped the `DNSServer.name == hostname` arm from `_find_appliance_dependents`. Match is now FK OR `host` only; comment block explains why `name` is unsafe (operator-controlled label, not a hostname guarantee).

File: backend/app/api/v1/appliance/supervisor.py [hunk 3]
```diff
@@ -1972,6 +1981,121 @@ class DeleteApplianceRequest(BaseModel):
...
+
+    dhcp_rows = (
+        (
+            await db.execute(
+                select(DHCPServer).where(
+                    or_(
+                        DHCPServer.appliance_id == appliance.id,
+                        DHCPServer.host == appliance.hostname,
+                        DHCPServer.name == appliance.hostname,
+                    )
```
[the remark this answers is not in the document]
Comment (mzac) [reply backend/app/api/v1/appliance/supervisor.py:2059]: Fixed in 2c96105 — same fix as the DNS thread. `DHCPServer.name` is unique + operator-set; matching against it risks deleting unrelated rows. Now matches on FK OR `host` only.

File: frontend/src/components/TimezoneSection.tsx [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+
+/**
+ * Appliance timezone control (issue #165).
+ *
+ * Settings → NTP tab uses this. The picker is a text input backed by
+ * a ``<datalist>`` of every IANA tz name the browser reports via
+ * ``Intl.supportedValuesOf("timeZone")`` — gives auto-complete
+ * without committing to a long ``<select>`` UX. Empty string clears
+ * the override (host falls back to install-time default).
+ *
```
[the remark this answers is not in the document]
Comment (mzac) [reply frontend/src/components/TimezoneSection.tsx:23]: Fixed in 2c96105 — both the form copy and the saved-confirmation note now accurately describe the behaviour: clearing the field stops further timezone changes from being pushed but does NOT revert the host's tz. To revert, the operator picks the desired value explicitly. Docstring also updated to call out the subtlety + cite this thread.

File: backend/app/services/ai/tools/webhooks.py [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+    module="webhooks",
+)
+async def list_webhooks(
+    db: AsyncSession, user: User, args: ListWebhooksArgs
+) -> list[dict[str, Any]]:
+    stmt = select(EventSubscription)
+    if args.enabled is not None:
+        stmt = stmt.where(EventSubscription.enabled.is_(args.enabled))
+    stmt = stmt.order_by(EventSubscription.name.asc()).limit(args.limit)
+    rows = (await db.execute(stmt)).scalars().all()
```
[the remark this answers is not in the document]
Comment (mzac) [reply backend/app/services/ai/tools/webhooks.py:83]: Fixed in 2c96105 — new `_superadmin_gate` helper (mirrors the pattern from `tools/pairing.py`) on both `list_webhooks` and `find_webhook_deliveries`. Non-superadmin users get an explanatory error instead of webhook URLs / delivery state.

File: backend/app/services/ai/tools/webhooks.py [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+) -> list[dict[str, Any]]:
+    stmt = select(EventOutbox)
+    if args.subscription_id:
+        stmt = stmt.where(EventOutbox.subscription_id == args.subscription_id)
+    if args.state:
+        stmt = stmt.where(EventOutbox.state == args.state)
+    if args.event_type:
+        stmt = stmt.where(EventOutbox.event_type == args.event_type)
+    stmt = stmt.order_by(EventOutbox.created_at.desc()).limit(args.limit)
+    rows = (await db.execute(stmt)).scalars().all()
```
[the remark this answers is not in the document]
Comment (mzac) [reply backend/app/services/ai/tools/webhooks.py:186]: Fixed in 2c96105 alongside the `list_webhooks` gate. Same `_superadmin_gate` helper applied to `find_webhook_deliveries`.

File: backend/app/services/ai/tools/conformity.py [hunk 7]
```diff
@@ -0,0 +1,240 @@
...
+@register_tool(
+    name="get_conformity_summary",
+    description=(
+        "Per-framework rollup of policy + result counts. Returns a "
+        "list of ``{framework, policy_count, enabled_count, "
+        "pass_count, fail_count, warn_count, not_applicable_count}``. "
+        "Counts only the MOST-RECENT result per (policy × resource) "
+        "pair so a failing run two days ago doesn't double-count "
+        "against today's pass. Useful for 'are we PCI-clean today?' "
+        "rollups. Read-only."
```
[the remark this answers is not in the document]
Comment (mzac) [reply backend/app/services/ai/tools/conformity.py]: Fixed in 2c96105 — re-wrote the tool description to honestly describe what the rollup is. The evaluator UPSERTs results per pass keyed on policy + resource, so most stale rows have been overwritten by current ones; rows for resources that no longer exist linger as a trailing-edge of the eval-interval window. Operators wanting point-in-time precision are directed at find_conformity_results.
